icrawler/downloader.py

```diff
@@ -115,12 +115,11 @@ def download(self,
 
         if not overwrite:
             with self.lock:
-                self.fetched_num += 1
                 filename = self.get_filename(task, default_ext)
                 if self.storage.exists(filename):
                     self.logger.info('skip downloading file %s', filename)
+                    self.fetched_num += 1
```

Comment:
The behavior of +1 followed by -1 is due to the implementation of `get_filename`, it depends on the counter to get the filename. We need to increase the counter first to check whether the filename of `fetched_num + 1` exists. (See also L144)
But I agree that current implementation is confusing anyway and may be improved.

Reply:
If the filename does not exist, we need to decrease the counter since it will be increased after the file is downloaded. Confusing indeed...